auth/src/VSCodeAzureSubscriptionProvider.ts

```diff
@@ -237,15 +242,21 @@ export class VSCodeAzureSubscriptionProvider extends vscode.Disposable implement
      *
      * @returns A client, the credential used by the client, and the authentication function
      */
-    private async getSubscriptionClient(tenantId?: string, scopes?: string[]): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {
+    private async getSubscriptionClient(tenantId?: string, scopes?: string[], session?: vscode.AuthenticationSession, account?: vscode.AuthenticationSessionAccountInformation): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {

```

Comment:
I think we should remove the session parameter to keep it simpler and always fetch the session within this method.

```suggestion
    private async getSubscriptionClient(tenantId?: string, scopes?: string[], account?: vscode.AuthenticationSessionAccountInformation): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {
```

And when we do this I think we can keep this check:
```ts
if (!session) {
    throw new NotSignedInError();
}
```

where it was before.